src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -514,15 +515,60 @@ void Terminal::SelectHyperlink(const SearchDirection dir)
                 searchArea = Viewport::FromDimensions(searchStart, { searchEnd.x + 1, searchEnd.y + 1 });
             }
         }
+    }
 
-        // 1.C) Nothing was found. Bail!
-        if (!result.has_value())
+    // 2. We found a hyperlink from the pattern tree. Look for embedded hyperlinks too!
+    // Use the result (if one was found) to narrow down the search.
+    searchStart = dir == SearchDirection::Forward ?
+                      _selection->start :
+                      (result ? result->second : bufferSize.Origin());
+    searchEnd = dir == SearchDirection::Forward ?
+                    (result ? result->first : buffer.GetLastNonSpaceCharacter()) :
+                    _selection->start;
+    auto iter = buffer.GetCellDataAt(dir == SearchDirection::Forward ? searchStart : searchEnd);
+    while (dir == SearchDirection::Forward ? iter.Pos() < searchEnd : iter.Pos() > searchStart)
+    {
+        // Don't let us select the same hyperlink again
+        if (iter.Pos() < _selection->start || iter.Pos() > _selection->end)
         {
-            return;
+            if (auto attr = iter->TextAttr(); attr.IsHyperlink())

```

Comment:
qq: do we have a way to iterate by attribute instead of by cell?

Reply:
Sorta, but it makes things more complicated and it's not really worth it imo.

We could use `row.Attributes().runs()` kinda like `TextBuffer::ClearMarksInRange()` here:
https://github.com/microsoft/terminal/blob/7423dd3b2a8726631f1c4c0a0c7b5f9fcd7def08/src/buffer/out/textBuffer.cpp#L3462-L3479

So we'd be able to get the length of each attribute run, but the combination of search direction (searching backwards with this system is hard) and support for wrapped hyperlinks makes this more complicated than it's worth, imo.